Automattic/wp-calypso
# Fix unified-sidebar scrolling

After a lot of try & fail, I eventually:
- used vanilla JS no without passing styles through react to the sidebar component. This way we avoid rerendering the component while scrolling.
- had to manipulate the `secondary` area and not the sidebar itself as it is encapsulated over there

#### Changes proposed in this Pull Request

* Adds `...props` pass through to `Sidebar` component.
* Adds `window.addEventListener( 'scroll')` in `MySitesSidebarUnified` component so that we reposition (scroll the sidebar) like wp-admin behaves.

#### Testing instructions



* load calypso with `?flags=nav-unification`
* test that the sidebar behaves as in wp-admin, especially when the window height is shorter than the sidebar
* test that my sites, reader, me work as expected
* test the mobile view (sidebar full width)

Relates #46643
